javascript/src/html_processor.ts

```diff
@@ -332,95 +411,84 @@ export class HTMLProcessor {
     // Add a sentinel to help iterating.
     boundaries.push(text.length + 1);
 
-    this.splitTextNodes(textNodes, boundaries);
+    this.splitNodes(paragraph.nodes, boundaries);
     this.applyBlockStyle(paragraph.element);
   }
 
   /**
-   * Split {@link Text} nodes at the specified boundaries.
-   * @param textNodes A list of {@link Text}.
+   * Split {@link NodeOrText} at the specified boundaries.
+   * @param nodes A list of {@link NodeOrText}.
    * @param boundaries A list of indices of the text to split at.
    */
-  splitTextNodes(textNodes: Text[], boundaries: number[]): void {
+  splitNodes(nodes: NodeOrText[], boundaries: number[]): void {
     assert(boundaries.length > 0);
-    const textLen = textNodes.reduce(
-      (sum, node) => sum + (node.nodeValue ? node.nodeValue.length : 0),
-      0
-    );
+    assert(boundaries.every((x, i) => i === 0 || x > boundaries[i - 1]));
+    const textLen = nodes.reduce((sum, node) => sum + node.length, 0);
     // The last boundary must be a sentinel.
     assert(boundaries[boundaries.length - 1] > textLen);
 
+    // Distribute `boundaries` to `node.boundaries`.
```

Comment:
Probably `node.boundaries` is an outdated comment here since a `node` does not have the `boundaries` property.

Reply:
Done, thanks for catching this.